docs/src/main/asciidoc/spanner.adoc

```diff
@@ -25,51 +29,198 @@ dependencies {
 }
 ----
 
+This setup takes care of bringing in the latest compatible version of Cloud Java Spanner libraries as well.
+
+
+=== Configuration
+
+To setup Spring Data Spanner, you have to configure the following:
+
+* setup the connection details to Google Cloud Spanner
+* enable Spring Data Repositories (optional).
+
+==== Spanner settings
+
+You will need to use link:../spring-cloud-gcp-starters/spring-cloud-gcp-starter-data-spanner[Spring Boot Starter for Spring Data Spanner] to autoconfigure Google Cloud Spanner in your Spring application.
+It contains all the necessary setup that makes it easy to authenticate with your Google Cloud project.
+The following configuration options are available:
+
+|===
+| Name | Description | Optional | Default value
+| `spring.cloud.gcp.spanner.instance-id` | Spanner instance to use | No |
+| `spring.cloud.gcp.spanner.database` |
+Spanner database to use | No |
+| `spring.cloud.gcp.spanner.project-id` | GCP project ID where the Google Cloud Pub/Sub API
+is hosted, if different from the one in the <<spring-cloud-gcp-core,Spring Cloud GCP Core Module>>
+| Yes |
+| `spring.cloud.gcp.spanner.credentials.location` | OAuth2 credentials for authenticating with the
+Google Cloud Spanner API, if different from the ones in the
+<<spring-cloud-gcp-core,Spring Cloud GCP Core Module>> | Yes |
+| `spring.cloud.gcp.spanner.credentials.scopes` |
+https://developers.google.com/identity/protocols/googlescopes[OAuth2 scope] for Spring Cloud GCP
+Spanner credentials | Yes | https://www.googleapis.com/auth/spanner.data
+|===
+
+==== Repository settings
+
+Spring Data Repositories can be configured via the `@EnableSpannerRepositories` annotation on your main `@Configuration` class.
```

Comment:
Worth mentioning that the annotation is automatically enabled using auto configuration.